### PR TITLE
enabled scrambling of data

### PR DESCRIPTION
**Pull request by ajaygargdev**

### CODE REVIEW

File: Solution/src/Kmd.Momentum.Mea.Common/Attributes/ScrambleDataAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+
+namespace Kmd.Momentum.Mea.Common.Attributes
```
Comment (amitxagarwal) [inline Solution/src/Kmd.Momentum.Mea.Common/Attributes/ScrambleDataAttribute.cs:3]: why these changes reflecting here? You should take the latest code from Dev and put the only changes needed in this bug fix.

File: Solution/src/Kmd.Momentum.Mea.Common/Attributes/ScrambleDataAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+{
+
+    [AttributeUsage(AttributeTargets.Property, AllowMultiple = false, Inherited = false)]
+    public class ScrambleDataAttribute : Attribute
```
Comment (amitxagarwal) [inline Solution/src/Kmd.Momentum.Mea.Common/Attributes/ScrambleDataAttribute.cs:7]: No need in this PR

File: Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Linq;
+using System.Threading.Tasks;
+using Serilog;
+
+namespace Kmd.Momentum.Mea.Common.Middleware
```
Comment (amitxagarwal) [inline Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs:12]: again no need in this PR

File: Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+{
+    public class ScrambleDataMiddleware
+    {
+        readonly RequestDelegate _next;
+
+        public ScrambleDataMiddleware(RequestDelegate next)
+        {
+            _next = next ?? throw new ArgumentNullException(nameof(next));
+
```
Comment (amitxagarwal) [inline Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs:21]: extra line

File: Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+            }
+            catch (Exception ex)
+            {
+                Log.ForContext("CorrelationId", httpContext.TraceIdentifier)
+                .Error("An error occurred while doing scramble" + ex.Message);
+            }
+            finally
+            {
+
+                httpContext.Response.Body = originBody;
```
Comment (amitxagarwal) [inline Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs:96]: extra line

File: Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+            if (httpContext == null) throw new ArgumentNullException(nameof(httpContext));
+
+            return InvokeInner(httpContext);
+        }
+
+        private async Task InvokeInner(HttpContext httpContext)
+        {
+            var originBody = httpContext.Response.Body;
+            try
+            {
```
Comment (amitxagarwal) [inline Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs:35]: put some logs in try block as well.

File: Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+                httpContext.Response.Body = originBody;
+            }
+        }
+
+        private IReadOnlyCollection<PropertyInfo> GetScrambledProperties(Type responseModelType)
+        {
+            return responseModelType.GetProperties().Where(p => p.CustomAttributes.ToList().Where(q => q.AttributeType.Name == "ScrambleDataAttribute").Any()).Where(p => p.PropertyType == typeof(Guid) || p.PropertyType == typeof(string)).ToList();
+
```
Comment (amitxagarwal) [inline Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs:103]: ?

File: Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs (hunk 4 above)
Comment (ajaygargdev) [reply Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs]: removed

File: Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs (hunk 5 above)
Comment (ajaygargdev) [reply Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs:127]: removed

File: Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs (hunk 6 above)
Comment (ajaygargdev) [reply Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs:34]: added logs

File: Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs (hunk 7 above)
Comment (ajaygargdev) [reply Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs]: removed extra line

File: Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+                    httpContext.Response.Body = responseBodyStream;
+
+                    await _next(httpContext);
+                }
+
+                responseBodyStream.Seek(0, SeekOrigin.Begin);
+                var responseBody = new StreamReader(responseBodyStream).ReadToEnd();
+                var _obj = JsonConvert.DeserializeObject<JToken>(responseBody);
+
+                Log.ForContext("CorrelationId", httpContext.TraceIdentifier)
```
Comment (ExploreMyCode) [inline Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs:49]: Do we need to write the debug log each time? Need to know more about this.

File: Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Linq;
+using System.Threading.Tasks;
+using Serilog;
+
+namespace Kmd.Momentum.Mea.Common.Middleware
```
Comment (ExploreMyCode) [inline Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs:12]: Very complex logic going on. I need your time to understand. Let us chat on this tomorrow.

File: Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs (hunk 8 above)
Comment (ajaygargdev) [reply Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs:49]: It will help to trace code if any issue comes in this code.

File: Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+                    if (responseModelType != null)
+                    {
+                        var resultProperty = responseModelType.GetProperties().Where(p => p.Name == "Result").FirstOrDefault();
+                        if (resultProperty != null && _obj["result"] != null && _obj["result"].Count() > 0)
+                        {
+                            var _modelTypeString = resultProperty.PropertyType.ToString();
+                            var _modelName = GetModelName(_modelTypeString);
+                            var _modelType = GetModelType(_modelName);
+
+                            var _scrambledProperties = GetScrambledProperties(_modelType);
```
Comment (amitxagarwal) [inline Solution/src/Kmd.Momentum.Mea.Common/Middleware/ScrambleDataMiddleware.cs:68]: what this part is for? Just stick to the fact that we are not going to scramble anything which is not under results.
